### PR TITLE
Add tranlucent no depth blending mode

### LINKED ISSUE #3343: Alpha doesn't work with depth setting in isosurface rendering
## 🐛 Bug

As @kevinyamauchi knows, I absolutely love his work in #3181. However, it has also ruined my workflow. :joy: 

Specifically, the way it currently works, *nothing* behind an object is currently rendered, *even if that object is transparent*. Here is code to reproduce the example:

```python
import napari
import numpy as np
from skimage import data, filters, morphology, segmentation
from scipy import ndimage as ndi

cells3d = data.cells3d()

viewer = napari.view_image(
    cells3d, channel_axis=1, name=['membranes', 'nuclei'], ndisplay=3
)
membrane, nuclei = cells3d.transpose((1, 0, 2, 3)) / np.max(cells3d)
edges = filters.scharr(nuclei)
denoised = ndi.median_filter(nuclei, size=3)
thresholded = denoised > filters.threshold_li(denoised)
cleaned = morphology.remove_small_objects(
    morphology.remove_small_holes(thresholded, 20**3),
    20**3,
)
maxima = ndi.label(morphology.local_maxima(filters.gaussian(nuclei, sigma=10)))[0]
markers_big = morphology.dilation(maxima, morphology.ball(5))

segmented = segmentation.watershed(
    edges,
    markers_big,
    mask=cleaned,
)

labels_layer = viewer.add_labels(segmented)
points_layer = viewer.add_points(np.transpose(np.nonzero(maxima)), size=2)

napari.run()
```

This results in the following display now, even on low opacity for the labels layer:

pts0

The points are actually there, but can only be seen if the Labels layer is made invisible:

pts1

Whereas it used to look like this:

pts2

It was always weird that the points would be "on top" of any previous layers, and messed with our perspective, but unfortunately I think the current behaviour is worse, in that transparency gives the impression that there is nothing "behind" a surface when in fact there is.

In the worst case, we can have a switch for setting depth or not during rendering.

### PR DESCRIPTION
**Pull request by kevinyamauchi**
# Description
This PR adds a blending mode that is the same as translucent, but without depth checking.

## Type of change
- [x] Bug-fix (non-breaking change which fixes an issue)

# References
Closes #3343 , supercedes #3395 

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Comment by kevinyamauchi** [comment]
Here's a screencap of the new blending mode.

https://user-images.githubusercontent.com/1120672/134046122-ac370903-f242-4ae0-baf6-c34aecb1684a.mov

```python
import napari
import numpy as np

data = np.zeros((100, 100, 100))
data[30:70, 30:70, 30:70] = 1

viewer = napari.view_image(data, rendering='iso')
viewer.add_points([50, 50, 50])

viewer.dims.ndisplay = 3

napari.run()

```

**Comment by kevinyamauchi** [comment]
@jni , would you mind taking this for a spin and seeing if this resolves your issue?

**Comment by Carreau** [comment]
Tests should be fixed by #3397

**Comment by Carreau** [comment]
I've rebased and force-push to see if the test are passing now that they are fixed on the main branch.

**Comment by kevinyamauchi** [comment]
Thanks, @Carreau !

File: napari/_vispy/utils_gl.py [hunk 1]
```diff
@@ -104,3 +104,38 @@ def fix_data_dtype(data):
...
+#         blend=True,
+#         blend_func=('src_alpha', 'one_minus_src_alpha', 'zero', 'one'),
+#     ),
+#     'additive': dict(
+#         depth_test=False,
+#         cull_face=False,
+#         blend=True,
+#         blend_func=('src_alpha', 'one'),
+#     ),
+# }
```
Comment (Carreau) [inline napari/_vispy/utils_gl.py]:
```suggestion
```

Beyond removing this commented section, the PR looks good to me, but I'll wait for someone more familiar to reviw.

**Review by jni — APPROVED** [review]
Yup, I can confirm that this resolves my issue @kevinyamauchi! 🙏🙏🙏

**Comment by jni** [comment]
I think my last commit will fix black. I'll merge this after successful tests if no one objects!

**Pull request merged**